include/gamepad/button.hpp

```diff
@@ -103,10 +111,50 @@ class Button {
          *   // Use a function...
          *   gamepad::master.A.onShortRelease("raiseLiftOneLevel", raiseLiftOneLevel);
          *   // ...or a lambda
-         *   gamepad::master.B.onShortRelease("intakeOnePicce", []() { intake.move_relative(600, 100); });
+         *   gamepad::master.B.onShortRelease("intakeOnePiece", []() { intake.move_relative(600, 100); });
          * @endcode
          */
         bool onShortRelease(std::string listenerName, std::function<void(void)> func) const;
+        /**
+         * @brief Register a function to run when the button is long released.
+         * By default, longRelease will fire when the button has been released after 500ms, this threshold can be
+         * adjusted by changing long_press_threshold.
+         *
+         * @param listenerName The name of the listener, this must be a unique name
+         * @param func The function to run when the button is long released, the function MUST NOT block
+         * @return true The listener was successfully registered
+         * @return false The listener was not successfully registered (there is already a listener with this name)
+         *
+         * @b Example:
+         * @code {.cpp}
+         *   // Use a function...
+         *   gamepad::master.A.onLongRelease("moveLiftToGround", moveLiftToGround);
+         *   // ...or a lambda
+         *   gamepad::master.B.onLongRelease("spinIntake", []() { intake.move(127); });
+         * @endcode
+         *
+         */
+        bool onLongRelease(std::string listenerName, std::function<void(void)> func) const;
+        /**
+         * @brief Register a function to run periodically after its been held
+         * By default repeatPress will start repeating after 500ms and repeat every 50ms, this can be changed by
+         * adjusting long_press_threshold and repeat_cooldown respectivly
```

Comment:
```suggestion
         *
         * By default repeatPress will start repeating after 500ms and repeat every 50ms, this can be changed by
         * adjusting long_press_threshold and repeat_cooldown respectively
```